AIX agent compilation is failing

|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| v4.6.0-alpha1 | Compilation - Packages generation | Agent | Packages/Sources | AIX 6.1 TL9 |


# Description

During https://github.com/wazuh/wazuh-jenkins/issues/5594, it was found that AIX agent cannot be compiled due to the next error
```
18:38:57  shared/sym_load.c: In function 'so_check_module_loaded':
18:38:57  shared/sym_load.c:61:30: error: 'RTLD_NOLOAD' undeclared (first use in this function)
18:38:57       return dlopen(file_name, RTLD_NOLOAD | RTLD_LAZY);
18:38:57                                ^~~~~~~~~~~
18:38:57  shared/sym_load.c:61:30: note: each undeclared identifier is reported only once for each function it appears in
18:38:57  shared/sym_load.c:63:1: warning: control reaches end of non-void function [-Wreturn-type]
18:38:57   }
18:38:57   ^
18:38:57  gmake[1]: *** [Makefile:1567: shared/sym_load.o] Error 1
18:38:57  gmake: *** [Makefile:790: agent] Error 2
```

Jenkins job: https://ci.wazuh.info/job/Packages_builder_special/794

--- a/src/headers/sym_load.h
+++ b/src/headers/sym_load.h
@@ -22,7 +22,8 @@ void* so_get_module_handle_on_path(const char *path, const char *so);
 void* so_get_module_handle(const char *so);
 /**
  * @brief Check if a module/library is already loaded. Must call a corresponding so_free_library()
- * if not used in WIN32.
+ * if not used in WIN32. If RTLD_NOLOAD isn't found in the system,
+ * the behavior is the same than so_get_module_handle().
  *
  * @param so The name of the module/library to check.
  * @return void* A handle to module if it is already loaded, NULL otherwise.

--- a/src/shared/sym_load.c
+++ b/src/shared/sym_load.c
@@ -1,6 +1,14 @@
 #include <stdio.h>
 #include "sym_load.h"
 
+#ifndef WIN32
+#ifdef RTLD_NOLOAD
+#define W_RTLD_NOLOAD RTLD_NOLOAD
+#else
+#define W_RTLD_NOLOAD 0x0
+#endif // RTLD_NOLOAD
+#endif // WIN32
+
 void* so_get_module_handle_on_path(const char *path, const char *so){
 #ifdef WIN32
     char file_name[MAX_PATH] = { 0 };
@@ -58,7 +66,7 @@ void* so_check_module_loaded(const char *so){
 #else
     snprintf(file_name, 4096-1, "lib%s.so", so);
 #endif
-    return dlopen(file_name, RTLD_NOLOAD | RTLD_LAZY);
+    return dlopen(file_name, W_RTLD_NOLOAD | RTLD_LAZY);
 #endif
 }
 

